jicofo-common/src/main/kotlin/org/jitsi/jicofo/xmpp/jingle/JingleRequestHandler.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Jicofo, the Jitsi Conference Focus.
+ *
+ * Copyright @ 2015-Present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.jicofo.xmpp.jingle
+
+import org.jitsi.xmpp.extensions.jingle.ContentPacketExtension
+import org.jitsi.xmpp.extensions.jingle.JingleIQ
+import org.jivesoftware.smack.packet.StanzaError
+
+/**
+ * Listener class notified about Jingle requests received during a session.
+ *
+ * @author Pawel Domas
+ */
+abstract class JingleRequestHandler {
```

Comment:
Is there a reason you turned this into an abstract class rather than keeping it an interface?

Reply:
Java and kotlin quirks. I removed a whole dummy implementation of this in java and wanted to use default methods, but to make kotlin defaults usable in java you need `@JvmDefault` which recquires a JVM param. Now that I think about it I can define a dummy impl in kotlin, so I'll change it back to an interface. I think we're run into this before.